lightning/src/ln/channelmanager.rs

```diff
@@ -3353,6 +3346,46 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 		for (source, payment_hash, reason) in timed_out_htlcs.drain(..) {
 			self.fail_htlc_backwards_internal(self.channel_state.lock().unwrap(), source, &payment_hash, reason);
 		}
+	}
+
+	/// Updates channel state to take note of transactions which were confirmed in the given block
+	/// at the given height.
+	///
+	/// Note that you must still call update_best_block with the block information here.
+	///
+	/// XXX: Note that in the case that a transaction which was provided via this method is
+	/// reorganized out of the best chain, the only current way to correctly handle the any
+	/// associated channel force-closure is to walk the chain back from the current tip with
+	/// repeated `block_disconnected` calls, followed by an `update_best_block` call.
```

Comment:
Is calling `update_best_block` requirement true ? `ChannelManager::block_disconnected` is calling `Channel::update_best_block` ?

Reply:
Mmm, docs unclear - I meant to go from the fork point to the new tip. In any case, I'm going to work on updating block_disconnected to build a cleaner API so I'll just fix it with that.

Reply:
Docs is better now by mentioning both reorg fork point and new best-chain tip as required called for `update_best_block`. Thanks